code/framework/encryption/src/main/java/io/cattle/platform/framework/encryption/impl/Aes128Encrypter.java

```diff
@@ -0,0 +1,72 @@
+package io.cattle.platform.framework.encryption.impl;
+
+import io.cattle.platform.framework.encryption.Encrypter;
+import io.cattle.platform.framework.encryption.EncryptionUtils;
+
+import java.security.Key;
+import java.security.SecureRandom;
+import javax.crypto.Cipher;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+import org.apache.commons.codec.binary.Hex;
+
+public class Aes128Encrypter extends Encrypter {
+    private String encryptionKey;
+
+    private Cipher encrypter;
+    private Cipher decrypter;
+    private static final String NAME = "AES128";
+    private Key key;
+    private SecureRandom rn;
+
+    public Aes128Encrypter() throws Exception {
+        rn = new SecureRandom();
+
+        encrypter = Cipher.getInstance("AES/CBC/PKCS5PADDING", "SunJCE");
+        try {
+            encryptionKey = EncryptionUtils.getKeyFromFile(NAME + "Key");
+        } catch (NullPointerException e) {
```

Comment:
Even if you move it there, I don't think catching NPE would be a good idea. It would be cleaner if some exception indicating that the key doesn't exist, is thrown by getKeyFromFile method. Or simply make it return null when the key can't be found. And just validate if encryptionKey is null. If its null, create a new one.